### ISSUE TITLE
Version 2+: `aplpy.FITSFigure` is not behaving properly

### ISSUE THREAD

**Issue by JulienPeloton** [aplpy/aplpy#420]
Hi,

thanks for the great package. I recently upgraded APLpy from 1.1.1 to 2+, and I ran into trouble using `aplpy.FITSFigure`. Simple things like opening a FITS file and displaying it does not work anymore (shows empty plots): 

```python
# This used to display correctly images
# in version 1.1.1
with fits.open(io.BytesIO(f.read())) as hdul:
  fig = ...
  subplot = ...
  ffig = aplpy.FITSFigure(
    hdul[0], figure=fig, subplot=subplot, **kwargs)
```

<img width="269" alt="screenshot 2019-02-18 at 23 21 51" src="https://user-images.githubusercontent.com/20426972/52979003-031fa680-33d4-11e9-9b1d-3770d02fd303.png">

If I then try to apply methods, such as `ffig.show_grayscale(stretch='arcsinh')`, it throws an error:

```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-21-40670a0835cd> in <module>
      8         o = plot_cutout(
      9             alert[cutout]["stampData"],
---> 10             fig=fig, subplot=(3, 3, 3*index_alert+index_cutout+1))
     11         if index_alert == 0:
     12             o.set_title(cutout)

<ipython-input-20-06d8a422c886> in plot_cutout(stamp, fig, subplot, **kwargs)
     21             ffig = aplpy.FITSFigure(
     22                 hdul[0], figure=fig, subplot=subplot, **kwargs)
---> 23             ffig.show_grayscale(stretch='arcsinh')
     24     return ffig

/opt/conda/lib/python3.6/site-packages/astropy/utils/decorators.py in show_grayscale(self, vmin, vmid, vmax, pmin, pmax, stretch, exponent, invert, smooth, kernel, aspect, interpolation)
    840             name = func.__name__
    841 
--> 842         func = make_function_with_signature(func, name=name, **wrapped_args)
    843         func = functools.update_wrapper(func, wrapped, assigned=assigned,
    844                                         updated=updated)

/opt/conda/lib/python3.6/site-packages/aplpy/decorators.py in wrapper(*args, **kwargs)
     26 
     27         try:
---> 28             return f(*args, **kwargs)
     29         finally:
     30             mydata.nesting -= 1

/opt/conda/lib/python3.6/site-packages/aplpy/core.py in show_grayscale(self, vmin, vmid, vmax, pmin, pmax, stretch, exponent, invert, smooth, kernel, aspect, interpolation)
    596                              stretch=stretch, exponent=exponent, cmap=cmap,
    597                              smooth=smooth, kernel=kernel, aspect=aspect,
--> 598                              interpolation=interpolation)
    599 
    600     @auto_refresh

/opt/conda/lib/python3.6/site-packages/astropy/utils/decorators.py in show_colorscale(self, vmin, vmid, vmax, pmin, pmax, stretch, exponent, cmap, smooth, kernel, aspect, interpolation)
    840             name = func.__name__
    841 
--> 842         func = make_function_with_signature(func, name=name, **wrapped_args)
    843         func = functools.update_wrapper(func, wrapped, assigned=assigned,
    844                                         updated=updated)

/opt/conda/lib/python3.6/site-packages/aplpy/decorators.py in wrapper(*args, **kwargs)
     26 
     27         try:
---> 28             return f(*args, **kwargs)
     29         finally:
     30             mydata.nesting -= 1

/opt/conda/lib/python3.6/site-packages/aplpy/core.py in show_colorscale(self, vmin, vmid, vmax, pmin, pmax, stretch, exponent, cmap, smooth, kernel, aspect, interpolation)
    697             stretch = 'asinh'
    698         normalizer = simple_norm(self._data, stretch=stretch, power=exponent,
--> 699                                  asinh_a=vmid, min_cut=vmin, max_cut=vmax)
    700 
    701         # Adjust vmin/vmax if auto

/opt/conda/lib/python3.6/site-packages/astropy/visualization/mpl_normalize.py in simple_norm(data, stretch, power, asinh_a, min_cut, max_cut, min_percent, max_percent, percent, clip)
    228     vmin, vmax = interval.get_limits(data)
    229 
--> 230     return ImageNormalize(vmin=vmin, vmax=vmax, stretch=stretch, clip=clip)
    231 
    232 

/opt/conda/lib/python3.6/site-packages/astropy/visualization/mpl_normalize.py in __init__(self, data, interval, vmin, vmax, stretch, clip)
     83         self.interval = interval
     84 
---> 85         self.inverse_stretch = stretch.inverse
     86         self.clip = clip
     87 

/opt/conda/lib/python3.6/site-packages/astropy/visualization/stretch.py in inverse(self)
    339     def inverse(self):
    340         """A stretch object that performs the inverse operation."""
--> 341         return SinhStretch(a=1. / np.arcsinh(1. / self.a))
    342 
    343 

TypeError: unsupported operand type(s) for /: 'float' and 'NoneType'
```

Any ideas? Thanks!

**Comment by astrofrog** [comment]
@JulienPeloton - thanks for the report, there is indeed a bug in the arcsinh stretch. Can you try not specifying a stretch to see if it works fine otherwise? I'll fix the stretch issue later today.

**Comment by JulienPeloton** [comment]
Thanks @astrofrog for the quick follow-up. Without applying a stretch, `FITSFigure` returns a figure with empty data (as shown above). I'm using python 3.7.0, aplpy 2.0.2, and astropy 3.1.1 if that can help.

**Comment by astrofrog** [comment]
@JulienPeloton - it is normal for the figure to be empty after calling just ``FITSFigure`` - you need to call at least ``ffig.show_grayscale()`` (with no arguments for no stretch) afterwards. Is that what you tried when you got the empty figure?

**Comment by JulienPeloton** [comment]
Oh, right! I was just calling `pyplot.show()`.

I confirm that calling `ffig.show_grayscale()` displays the figure correctly.

**Comment by astrofrog** [comment]
@JulienPeloton - I've just released 2.0.3 which fixes the arcsinh issue.

**Comment by JulienPeloton** [comment]
@astrofrog great, 2.0.3 tested and approved! 
Thanks for the quick fix.
